src/DynamoCore/Models/AnnotationModel.cs

```diff
@@ -0,0 +1,379 @@
+            var xgroup = SelectedModels.OrderBy(x => x.X).ToList();
+            var ygroup = SelectedModels.OrderBy(x => x.Y).ToList();
+          
+            return Tuple.Create(xgroup.Last().Width, ygroup.Last().Height);
+        }
+        
+        /// <summary>
+        /// Deserializes the model guids from XML
+        /// and creates group on those model guids.
+        /// </summary>
+        private void DeserializeGroup()
+        {
+            var listOfModels = new List<ModelBase>();
+            foreach (var objGuid in modelGuids.Split(','))
+            {
+                Guid result;
+                if (Guid.TryParse(objGuid, out result))
+                    if (SelectedModels != null)
+                    {
+                        var model = SelectedModels.FirstOrDefault(x => x.GUID == result);
+                        listOfModels.Add(model);
+                    }
+            }
+            SelectedModels = listOfModels;           
+        }
+
+        #region Serialization/Deserialization Methods
+
+        protected override void SerializeCore(XmlElement element, SaveContext context)
+        {            
+            XmlElementHelper helper = new XmlElementHelper(element);
+            helper.SetAttribute("guid", this.GUID);
+            helper.SetAttribute("text", this.AnnotationText);
+            helper.SetAttribute("left", this.Left);
+            helper.SetAttribute("top", this.Top);
+            helper.SetAttribute("width", this.Width);
+            helper.SetAttribute("height", this.Height);
+            helper.SetAttribute("fontSize", this.FontSize);
+            helper.SetAttribute("initialTop", this.InitialTop);
+            helper.SetAttribute("initialHeight", this.InitialHeight);
+            helper.SetAttribute("annotationColor", (this.BackGroundColor == null ? "" : this.BackGroundColor.ToString()));
+            helper.SetAttribute("ModelGUIDs", string.Join(",", this.SelectedModels.Select(x => x.GUID)));           
+        }
+
+        protected override void DeserializeCore(XmlElement element, SaveContext context)
+        {            
+            XmlElementHelper helper = new XmlElementHelper(element);
+            this.GUID = helper.ReadGuid("guid", this.GUID);
+            this.AnnotationText = helper.ReadString("text", "<<Click to edit the grouping>>");
+            this.Left = helper.ReadDouble("left", 0.0);
+            this.Top = helper.ReadDouble("top", 0.0);
+            this.Width = helper.ReadDouble("width", 0.0);
+            this.Height = helper.ReadDouble("height", 0.0);
+            this.BackGroundColor = helper.ReadString("annotationColor", "");
+            this.FontSize = helper.ReadDouble("fontSize", 10.0);
+            this.InitialTop = helper.ReadDouble("initialTop", 0.0);
+            this.InitialHeight = helper.ReadDouble("initialHeight", 0.0);
+            modelGuids = helper.ReadString("ModelGUIDs", "");           
+            DeserializeGroup();           
```

Comment:
Defaults should be encoded in const fields.  This makes them easier to change then leaving them buried in a `DeserializeCore` method.